internal/transport/http2_client.go

```diff
@@ -1025,6 +1027,11 @@ func (t *http2Client) Close(err error) {
 	}
 	t.cancel()
 	t.conn.Close()
+	// Wait for the reader channel to close
```

Comment:
Comments should explain _why_, not _what_.  So this is `// Wait for the reader goroutine to exit to ensure all resources are cleaned up before Close can return.`.  Similar below.  Also make sure comments that are sentences are punctuated correctly (e.g. end with a period).

Reply:
Done